invenio_damap/generators.py

```diff
@@ -34,7 +34,9 @@ def default_namespaced_id_generator(
         for ra in user.remote_accounts:
             for token in ra.remote_tokens:
                 dec_token = jwt.decode(
-                    token, current_app.config["SECRET_KEY"], algorithms=["RS256"]
+                    token,
+                    current_app.config["SECRET_KEY"],
+                    algorithms=current_app.config["DAMAP_JWT_DECODE_ALGORITHMS"],
```

Comment:
this won't work.
note that here we're talking about OIDC tokens that come from the authentication pipeline (e.g. Keycloak instances registered via Invenio-OAuthClient), and not the JWTs that are exchanged between Invenio and Damap (which would use symmetric pre-shared keys).

each entry in the `ra.remote_tokens` list is an object of type `invenio_oauthclient.models.RemoteToken`.
`jwt.decode()` expects the token to be `str` or `bytes` (like `token.access_token`).

this assumes the `token.access_token` to be a JWT, which is the case for access tokens provided by Keycloak.
however, other OIDC providers - such as SATOSA - may deliver random strings as access tokens.
those can only be used as bearer tokens to access OIDC endpoints (like the `userinfo` endpoint), but they cannot be decoded because they are just random strings.
in case that an instance has set up such an OIDC provider via Invenio-OAuthClient, this logic here will likely break – best to wrap the decoding in a `try/except`.

the key used for signing the token/verifying the signature/decoding the token isn't the Invenio app's `SECRET_KEY`.
generally, OIDC provider applications like Keycloak use **asymmetric** keys for their tokens, and advertise the necessary information under their JWKS endpoint (e.g. https://auth.researchdata.tuwien.ac.at/oidc/jwks).

very likely these tokens are also relatively short-lived (in the order of minutes) and by the time this code gets to try and verify their signature, it has probably expired a long time ago.

given that the `user.remote_accounts[].remote_tokens[]` should only be populated by `Invenio-OAuthClient` with (once) valid tokens anyway, it should be sufficient to trust the system and skip the signature validation here.
i.e.
```python
try: 
  dec_token = jwt.decode(token.access_token, options={"verify_signature": False})
except:
  continue
```